x/concentrated-liquidity/tick.go

```diff
@@ -405,22 +413,42 @@ func (k Keeper) GetTickLiquidityNetInDirection(ctx sdk.Context, poolId uint64, t
 	for ; iterator.Valid(); iterator.Next() {
 		tickIndex, err := types.TickIndexFromBytes(iterator.Key())
 		if err != nil {
-			return []queryproto.TickLiquidityNet{}, err
+			return []queryproto.TickLiquidityNet{}, 0, sdk.Dec{}, err
 		}
 
 		tickStruct, err := ParseTickFromBz(iterator.Value())
 		if err != nil {
-			return []queryproto.TickLiquidityNet{}, err
+			return []queryproto.TickLiquidityNet{}, 0, sdk.Dec{}, err
 		}
 
 		liquidityDepth := queryproto.TickLiquidityNet{
 			LiquidityNet: tickStruct.LiquidityNet,
 			TickIndex:    tickIndex,
 		}
+
+		// If we are using the current tick, there is no need to accumulate liquidity values here since we store current tick liquidity in state.
+		if startTick == p.GetCurrentTick() {
+			liquidityDepths = append(liquidityDepths, liquidityDepth)
+			continue
+		}
+
+		// Otherwise, depending on the swap direction, we need to accumulate liquidity values to determine the liquidity at the start tick.
+		if zeroForOne {
+			if liquidityDepth.TickIndex < currentTick {
+				if liquidityDepth.LiquidityNet.IsPositive() {
```

Comment:
What is the reason we only do operation on positive case here?

Reply:
Because I am wrong, and you are right. Let me add a test case to prove this and remove the positive check

Reply:
Fixed here https://github.com/osmosis-labs/osmosis/pull/5466/commits/81f34f763cf1cd73307905a2e728f1229e450a47

Also added a test that would have caught this issue. Thanks for flagging!